app/src/main/java/lt/getpet/getpet/OnboardingActivity.kt

```diff
@@ -0,0 +1,59 @@
+package lt.getpet.getpet
+
+import android.os.Bundle
+import kotlinx.android.synthetic.main.activity_onboarding.*
+import lt.getpet.getpet.adapters.OnboardingAdapter
+import lt.getpet.getpet.navigation.NavigationManager
+import javax.inject.Inject
+import androidx.viewpager.widget.ViewPager
+import lt.getpet.getpet.preferences.AppPreferences
+
+
+class OnboardingActivity : BaseActivity() {
+
+    @Inject
+    lateinit var navigationManager: NavigationManager
+
+    @Inject
+    lateinit var appPreferences: AppPreferences
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_onboarding)
+
+        val adapter = OnboardingAdapter(supportFragmentManager)
+
+        onboarding_viewpager.adapter = adapter
+
+        indicator.setViewPager(onboarding_viewpager)
+
+        onboarding_viewpager.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
+            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
+            }
+
+            override fun onPageSelected(position: Int) {
+                if (position == adapter.count - 1) button_next.setText(getString(R.string.button_end_onboarding))
+                else button_next.setText(getString(R.string.button_next))
```

Comment:
I wouldn't recommend to omit curly braces around if. This leads to strange errors. It's personal choice. More information why https://stackoverflow.com/questions/8020228/is-it-ok-if-i-omit-curly-braces-in-java